### ISSUE TITLE
ability to clear headers on outbound requests

### ISSUE THREAD

**Issue by siggy** [linkerd/linkerd#1218]
A user reported not being able to proxy requests via Linkerd to Loggly:
https://stackoverflow.com/questions/43295644/node-global-tunnel-and-loggly-and-linkerd

It turns out Loggly returns a 400 when a request has headers with forward slashes in it:

normal request:
```bash
$ curl -H "foo: bar" http://logs-01.loggly.com/
Invalid API key
```

failed request:
```bash
$ curl -H "l5d-dst-logical: /svc/logs-01.loggly.com" http://logs-01.loggly.com/
<html><head><title>Bad Request</title></head><body><h1>400 Bad Request</h1></body></html>
```

While it's debatable whether refusing this class of requests is valid, it would still be helpful for linkerd to support proxying to Loggly. Additionally, there may be use cases where operators simply do not want to leak information from inside their clusters to the internet.

We note in the [linkerd documentation](https://linkerd.io/config/0.9.1/linkerd/index.html#informational-request-headers) that `Operators may opt to remove this header from responses sent to the outside world.` This implies setting up an additional service/nginx/proxy/something to strip headers on outbound requests. It would be more convenient to the operator for linkerd to handle this for them.

**Comment by adleong** [comment]
Relatedly, we could also strip off informational headers linkerd sends back on the response.  I'm not sure if there are any yet but https://github.com/linkerd/linkerd/pull/1185/files would add `l5d-success-class` to indicate response classification.

**Comment by olix0r** [comment]
Are slashes forbidden by the RFC? Is this just  a bug that we're not
properly encoding headers?

**Comment by olix0r** [comment]
@adleong I think it would make sense for what you describe to be rolled into the clearContext feature...

**Comment by olix0r** [comment]
No, slashes are _obviously_ allowed in http headers (e.g. text/html).  loggy is silly.

This does seem like a reasonable client feature.
